tasks/npm-publish-clean-typechain.ts

```diff
@@ -0,0 +1,20 @@
+import { subtask } from 'hardhat/config';
+import { TASK_COMPILE_SOLIDITY_COMPILE_JOBS } from 'hardhat/builtin-tasks/task-names';
+import fs from 'fs/promises';
+
+subtask(TASK_COMPILE_SOLIDITY_COMPILE_JOBS, 'Clean mocks from types if needed').setAction(async (taskArgs, { run }, runSuper) => {
+  const compileSolOutput = await runSuper(taskArgs);
+  if (!!process.env.PUBLISHING_NPM) {
+    console.log('🧹 Excluding mock types from typechain');
+    const typechainIndexBuffer = await fs.readFile('./typechained/index.ts');
+    const typechainIndex = typechainIndexBuffer.toString('utf-8');
+    let finalTypechainIndex = '';
+    typechainIndex.split(/\r?\n/).forEach((line) => {
+      if (line.indexOf('Mock') === -1) {
+        finalTypechainIndex += line + '\n';
+      }
+    });
```

Comment:
I think this is a little bit clearer:
```suggestion
    const finalTypechainIndex = typechainIndexBuffer
        .toString('utf-8')
        .split(/\r?\n/)
        .filter(line => !line.includes('Mock'))
        .join('\n')
```

Reply:
Yeap. 100% more clear. Will commit suggestion and test it out.